### DISCUSSION TITLE
LED indicator when charing is stopped due to overheating

### DISCUSSION THREAD

**Discussion by devklepacki** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1014] [Ideas]
As in the title, my common issue with Aldente is that I think my battery is charging while in fact charging is stopped due to the battery exceeding the allowed temperature.

What'd love to have is something like the MagSafe LED indicator that's always off (as I have) but also very slowly blinking in orange (or anyhow) when charging is stopped due to overheating of the battery. 

I just need any kind of an indicator that my battery is in fact not charging.

**Comment by andrewcrook** [comment]
Well you can see from the menu icon, depending on how you set it up, but I expect you're asking for a hardware option like controlling MagSafe LEDs etc  If so, it depends if it is possible to gain access to that feature via an API. With MagSafe I am guessing probably not if the T2 chip is being used for that.

**Reply by andrewcrook** [reply]
Some of the limitations of Aldente are purely because of Apples hardware restrictions etc T2 chip. 
Such as controlling charging when shutdown or during booting process etc. 
Ironically, some of the older Intel hardware didn’t actually have some of these limitations.

**Reply by andrewcrook** [reply]
I am just thinking what could be used....

**Reply by devklepacki** [reply]
On the menu icon, I only have a current % value, so there's no indicator there. I also don't look there very often as I have menu bar hidden.

It's just to add a similar toggle as the ones below:

<img width="438" alt="image" src="https://github.com/davidwernhart/AlDente-Charge-Limiter/assets/20723654/95bd6ee2-d7fb-4805-9224-4bfbaad6ed07">

to Heat Protection section here:

<img width="463" alt="image" src="https://github.com/davidwernhart/AlDente-Charge-Limiter/assets/20723654/0ce089ec-f1d3-42a3-adb7-3f7bbea2836b">

With a label along the lines:

> Blink Magsafe LED Orange when activated (overrides LED Always Off setting)

That'd be a great indicator that a battery is in fact not charging, what happens quite often to me.

**Reply by andrewcrook** [reply]
I have Aldente setup like this  ( I am using pro version) then hide the system one

<img width="101" alt="Screenshot 2023-08-02 at 17 17 55" src="https://github.com/davidwernhart/AlDente-Charge-Limiter/assets/803618/01c39080-2c78-4b11-8716-1693fb4da1b5">


<img width="418" alt="Screenshot 2023-08-02 at 17 19 44" src="https://github.com/davidwernhart/AlDente-Charge-Limiter/assets/803618/33252aa9-29d8-42bc-bd27-cdbba5dec155">

**Reply by andrewcrook** [reply]
I can see something flashing or brighter might be a good idea

**Reply by devklepacki** [reply]
This wouldn't be that bad actually! But I try to limit the space taken by the icon as much as possible to keep it clean, and this doesn't happen that often, so the tradeoff of having more messier Menu Bar is not worth it for me 😅 .

But what's more important is that having this as the MagSafe LED indicator would allow **an indication that Heat Protection is activated even with the lid closed**. That's what troubling me. I close down the computer and set it off to charge, and after a while when I want to use it I notice it's barely charged.

**Reply by andrewcrook** [reply]
Out of interest, what sort of temperatures are you reaching on your machine and room temp?

**Reply by devklepacki** [reply]
I have the Heat Protection limit set to 35 degrees, and it's sometimes even over 30 degrees outside in the shade.

**Reply by devklepacki** [reply]
For sure https://github.com/davidwernhart/AlDente-Charge-Limiter/issues/630 would help immensely, but a little indicator would be useful anyway :)

**Comment by MatthiasKerbl** [comment]
> As in the title, my common issue with Aldente is that I think my battery is charging while in fact charging is stopped due to the battery exceeding the allowed temperature.
> 
> What'd love to have is something like the MagSafe LED indicator that's always off (as I have) but also very slowly blinking in orange (or anyhow) when charging is stopped due to overheating of the battery.
> 
> I just need any kind of an indicator that my battery is in fact not charging.

Hi @devklepacki,

I understand where your suggestion is coming from, however, I am afraid it is too niche of an issue and would make AlDente Pro unnecessarily complicated. For now, I would recommend increasing the Heat Protection temperature, if charging takes too long with Heat Protection in your case. Furthermore, we will try to implement a temperature range and/or add fan control which should also help with your issue.
I will leave this suggestion open here if other users want to share their feedback about this topic.

Best,
Matthias
